CONTRIBUTING.md

```diff
@@ -50,7 +50,7 @@ git clone git@github.com:determined-ai/determined.git
 
 - Go (>= 1.16)
 - Python (>= 3.6, < 3.8)
-- Node (>= 12)
+- Node (>= 12, < 16)
 - NPM (>= 6.12)
```

Comment:
npm usually comes bundled with node I've never mixed and matched these manually but to be safe let's also put an upper limit here.

the latest active lts comes with `Node.js 14.17.1	Fermium	2021-06-15	8.4.371.23	NPM: 6.14.13	83`

Reply:
I don't know that we need to upper-bound NPM until we actually have evidence it doesn't work on newer releases.(though `6.14.13` is what I used here)